src/System.Reflection.TypeExtensions/tests/EventInfoTests.cs

```diff
@@ -0,0 +1,184 @@
+            Assert.Equal(EventAttributes.None, eventInfo.Attributes);
+            Assert.False(eventInfo.IsSpecialName);
+        }
+
+        [Theory]
+        [InlineData("PublicEvent", typeof(VoidDelegate))]
+        [InlineData("PrivateEvent", typeof(VoidDelegate))]
+        [InlineData("ProtectedEvent", typeof(VoidDelegate))]
+        [InlineData("ProtectedInternalEvent", typeof(VoidDelegate))]
+        public void EventHandlerType(string name, Type expected)
+        {
+            EventInfo eventInfo = Helpers.GetEvent(typeof(EI_Class), name);
+            Assert.Equal(expected, eventInfo.EventHandlerType);
+        }
+
+        [Theory]
+        [InlineData("PublicEvent", "Void add_PublicEvent(System.Reflection.Tests.VoidDelegate)", "Void add_PublicEvent(System.Reflection.Tests.VoidDelegate)")]
+        [InlineData("PrivateEvent", null, "Void add_PrivateEvent(System.Reflection.Tests.VoidDelegate)")]
+        [InlineData("ProtectedEvent", null, "Void add_ProtectedEvent(System.Reflection.Tests.VoidDelegate)")]
+        [InlineData("ProtectedInternalEvent", null, "Void add_ProtectedInternalEvent(System.Reflection.Tests.VoidDelegate)")]
+        public void GetAddMethod(string name, string publicToString, string nonPublicToString)
+        {
+            EventInfo eventInfo = Helpers.GetEvent(typeof(EI_Class), name);
+            MethodInfo method = eventInfo.GetAddMethod();
+            Assert.Equal(publicToString == null, method == null);
+            Assert.Equal(publicToString, method?.ToString());
+
+            method = eventInfo.GetAddMethod(false);
+            Assert.Equal(publicToString == null, method == null);
+            Assert.Equal(publicToString, method?.ToString());
+
+            method = eventInfo.GetAddMethod(true);
+            Assert.NotNull(method);
+            Assert.Equal(nonPublicToString, method.ToString());
+        }
+
+        [Theory]
+        [InlineData("PublicEvent")]
+        [InlineData("PrivateEvent")]
+        [InlineData("ProtectedEvent")]
+        [InlineData("ProtectedInternalEvent")]
+        public void GetRaiseMethod(string name)
+        {
+            EventInfo eventInfo = Helpers.GetEvent(typeof(EI_Class), name);
+            Assert.Null(eventInfo.GetRaiseMethod());
+            Assert.Null(eventInfo.GetRaiseMethod(false));
+            Assert.Null(eventInfo.GetRaiseMethod(true));
+        }
+
+        [Theory]
+        [InlineData("PublicEvent", "Void remove_PublicEvent(System.Reflection.Tests.VoidDelegate)", "Void remove_PublicEvent(System.Reflection.Tests.VoidDelegate)")]
+        [InlineData("PrivateEvent", null, "Void remove_PrivateEvent(System.Reflection.Tests.VoidDelegate)")]
+        [InlineData("ProtectedEvent", null, "Void remove_ProtectedEvent(System.Reflection.Tests.VoidDelegate)")]
+        [InlineData("ProtectedInternalEvent", null, "Void remove_ProtectedInternalEvent(System.Reflection.Tests.VoidDelegate)")]
```

Comment:
We shouldn't need to include duplicate strings for public and non-public.  The argument to GetAdd/RemoveMethod controls whether a non-public method can be returned, so really we just need the string and a Boolean to indicate whether that string or null is expected.